src/sagemaker/jumpstart/utils.py

```diff
@@ -845,7 +857,22 @@ def get_jumpstart_model_id_version_from_resource_arn(
                 break
             model_version = model_version_from_tag
 
-    return model_id, model_version
+    for config_name_key in model_config_name_keys:
+        try:
+            config_name_key_from_tag = get_tag_value(config_name_key, list_tags_result)
+        except KeyError:
+            continue
+        if config_name_key_from_tag is not None:
+            if config_name is not None and config_name_key != config_name:
+                constants.JUMPSTART_LOGGER.warning(
+                    "Found multiple model config names tags on the following resource: %s",
+                    resource_arn
+                )
+                config_name = None
+                break
+            config_name = config_name_key_from_tag
+
```

Comment:
this logic seems to be repeated many times throughout, consider consolidating

Reply:
Yes extracted to a shared function